packages/local_auth/local_auth_darwin/darwin/local_auth_darwin/Sources/local_auth_darwin/LocalAuthPlugin.swift

```diff
@@ -0,0 +1,421 @@
+          let url = URL(string: UIApplication.openSettingsURLString)
+        {
+          let additionalAction = UIAlertAction(
+            title: openSettingsButtonTitle,
+            style: .default
+          ) { [weak self] action in
+            UIApplication.shared.open(url, options: [:], completionHandler: nil)
+            self?.handleResult(succeeded: false, completion: completion)
+          }
+          alert.addAction(additionalAction)
+        }
+        alert.present(on: controller, animated: true, completion: nil)
+      } else {
+        // TODO(stuartmorgan): Create a new error code for failure to show UI, and return it here.
+        self.handleResult(succeeded: false, completion: completion)
+      }
+    #endif
+  }
+
+  private func handleAuthReply(
+    success: Bool,
+    error: Error?,
+    options: AuthOptions,
+    strings: AuthStrings,
+    completion: @escaping (Result<AuthResultDetails, Error>) -> Void
+  ) {
+    if success {
+      handleResult(succeeded: true, completion: completion)
+      return
+    }
+
+    if let error = error as? NSError {
+      switch LAError.Code(rawValue: error.code) {
+      case .biometryNotAvailable,
+        .biometryNotEnrolled,
+        .biometryLockout,
+        .userFallback,
+        .passcodeNotSet,
+        .authenticationFailed:
+        handleError(error, options: options, strings: strings, completion: completion)
+      case .systemCancel:
+        if options.sticky {
+          lastCallState = StickyAuthState(
+            options: options,
+            strings: strings,
+            resultHandler: completion)
+        } else {
+          handleResult(succeeded: false, completion: completion)
+        }
+      default:
+        handleError(error, options: options, strings: strings, completion: completion)
+      }
+    } else {
+      // The Obj-C declaration of evaluatePolicy defines the callback type as NSError*, but the
+      // Swift version is (any Error)?, so provide a fallback in case somehow the type is not
+      // NSError.
+      // TODO(stuartmorgan): Add an "unknown error" enum option and return that here instead of
+      // failure.
+      completion(
+        .success(
+          AuthResultDetails(
+            result: .failure,
+            errorMessage: "Unknown error from evaluatePolicy",
+            errorDetails: error?.localizedDescription)
+        ))
+    }
+  }
+
+  private func handleResult(
+    succeeded: Bool, completion: @escaping (Result<AuthResultDetails, Error>) -> Void
+  ) {
+    completion(
+      .success(
+        AuthResultDetails(
+          result: succeeded ? .success : .failure,
+          errorMessage: nil,
+          errorDetails: nil)
+      ))
+  }
+
+  private func handleError(
+    _ authError: NSError,
+    options: AuthOptions,
+    strings: AuthStrings,
+    completion: @escaping (Result<AuthResultDetails, Error>) -> Void
+  ) {
+    var result = AuthResult.errorNotAvailable
```

Comment:
nit: can be `let result: AuthResult` (prefer immutability). This will fail to compile if any of the case below forget to set the result.

Reply:
Done. I didn't know Swift could do this to; I'm a big fan of the pattern in Dart.